### PR TITLE
Hotfix/routes and allowable areas

### PR DESCRIPTION
**Pull request by IanMayo**
Hotfix to resolve missing route markers & allowable routes.

### CODE REVIEW

File: packages/components/src/local/asset-icon/index.tsx [hunk 1]
```diff
@@ -36,17 +36,24 @@ export const AssetIcon: React.FC<PropTypes> = ({
...
-      type,
-      force,
-      visibleTo,
-      controlledBy,
-      condition,
-      status
-    })
-    setShowMapBar(true)
+    if (selectedAsset && selectedAsset.uniqid === uniqid) {
+      // ok, it's the current asset. clear it
```
Comment (foxleigh81) [inline packages/components/src/local/asset-icon/index.tsx]: Remember that I said comments shouldn't be conversational. This should be.

// Clear the current asset.

File: packages/components/src/local/mapping/index.tsx [hunk 2]
```diff
@@ -210,6 +190,7 @@ export const Mapping: React.FC<PropTypes> = ({
           origin: lastCell.name,
           travelMode: planningConstraints.travelMode
         }
+        console.log('new constraints', newP)
```
Comment (foxleigh81) [inline packages/components/src/local/mapping/index.tsx]: Did you mean to leave this in?

File: packages/components/src/local/mapping/index.tsx [hunk 3]
```diff
@@ -252,7 +233,9 @@ export const Mapping: React.FC<PropTypes> = ({
         for (let ctr = 0; ctr < plannedTurn.turnsVal; ctr++) {
           const step: RouteStep = { turn: ++turnStart, status: { state: status.name } }
           // store this step
-          store = routeAddStep(store, selectedAsset.uniqid, step)
+          if (selectedAsset) {
+            store = routeAddStep(store, selectedAsset.uniqid, step)
```
Comment (foxleigh81) [inline packages/components/src/local/mapping/index.tsx]: Be careful with circular references like this.  You might be safer doing `updatedStore = routeAddStep(etc...)` obviously don't forget to declare `updateStore` before you reference it.

Comment (foxleigh81) [reply packages/components/src/local/mapping/index.tsx]:
```suggestion
           // before conditional 
           let updatedStore
           
           // in conditional
            updatedStore = routeAddStep(store, selectedAsset.uniqid, step)
            
            // afterConditional
            setRouteStore(updatedStore)
```

Comment (IanMayo) [reply packages/components/src/local/mapping/index.tsx]: In the loop, we need to add the new step to the modified store:

 updatedStore = routeAddStep(***updatedStore***, selectedAsset.uniqid, step)

Comment (foxleigh81) [reply packages/components/src/local/mapping/index.tsx]: Leave it as it is then if that's the way it has to be. Maybe rename the variable though as there is a constant with the same name earlier in the file and it was momentarily confusing.

**Pull request merged**